x-pack/plugins/apm/server/routes/service_groups/get_service_groups.ts

```diff
@@ -5,23 +5,32 @@
  * 2.0.
  */
 
-import { SavedObjectsClientContract } from '@kbn/core/server';
+import {
+  IUiSettingsClient,
+  SavedObjectsClientContract,
+} from '@kbn/core/server';
+import { serviceGroupMaxNumberOfServices } from '@kbn/observability-plugin/common';
 import {
   ServiceGroup,
   SavedServiceGroup,
   APM_SERVICE_GROUP_SAVED_OBJECT_TYPE,
-  MAX_NUMBER_OF_SERVICE_GROUPS,
 } from '../../../common/service_groups';
 
 export async function getServiceGroups({
   savedObjectsClient,
+  uiSettingsClient,
 }: {
   savedObjectsClient: SavedObjectsClientContract;
+  uiSettingsClient: IUiSettingsClient;
 }): Promise<SavedServiceGroup[]> {
+  const maxNumberOfServices = await uiSettingsClient.get<number>(
+    serviceGroupMaxNumberOfServices
+  );
+
   const result = await savedObjectsClient.find<ServiceGroup>({
     type: APM_SERVICE_GROUP_SAVED_OBJECT_TYPE,
     page: 1,
-    perPage: MAX_NUMBER_OF_SERVICE_GROUPS,
+    perPage: maxNumberOfServices,
```

Comment:
Based on my knowledge a different constant should be used here for the service group pagination?

Reply:
It seems so, is this a bug atm?

Reply:
@dgieselaar do you know if there's any special reason why there's this limitation on the service groups?

Reply:
This should be `MAX_NUMBER_OF_SERVICE_GROUPS`. It's requesting the service groups (stored as saved objects), not services. You always need to tell the saved objects client how many items you are requesting, or you'd need to paginate through it. I think it is reasonable to set a maximum number for this reason, we currently don't really have the UI to support an infinite number of service groups. Does that answer all questions?

Reply:
I reverted back to using a constant for this one instead of using the advanced config